[core]/es_extended/client/main.lua

```diff
@@ -358,8 +358,11 @@ function StartServerSyncLoops()
 						else
 							if ammoCount ~= currentWeapon.Ammo then
 								currentWeapon.Ammo = ammoCount
-								TriggerServerEvent('esx:updateWeaponAmmo', weapon.name, ammoCount)
-							end
+                            Player(GetPlayerServerId(PlayerId())).state:set('ESX::WeaponUpdate', {
+								name = weapon.name,
+								ammo = ammoCount, 
+							    Random = GetGameTimer() + math.random(99999999, 999999999)
```

Comment:
What is this?

Reply:
state bags instead of triggering the server event way more optimized

Reply:
Sorry i mean this row: 
`Random = GetGameTimer() + math.random(99999999, 999999999)
 `

Reply:
LocalPlayer.state:set ??????????

Reply:
To what gellipapa said it doesnt pass half the time without to what Rav3n Said im not the brightest

Reply:
Sorry i don't understand what do you mean. Raven tell you you should use LocalPlayer in client side. Read FiveM docs how to works statebags.

Reply:
LocalPlayer.state:set('ESX::WeaponUpdate', {
								name = weapon.name,
								ammo = ammoCount, 
							    Random = GetGameTimer() + math.random(99999999, 999999999)

Reply:
fixed

Reply:
> fixed

Please use what Raven said or we will not accept it.

> LocalPlayer.state:set ??????????